Staircase.js

```diff
@@ -265,65 +296,107 @@ Staircase.prototype.getActive = function () {
+  this.checkErr.ARG('active', arguments, 1);
+  this.checkErr.UNDEFINED(this.stairs, stair);
   for (var i in this.stairs) {
     if (this.stairs[i].active) {
       return i;
     }
   }
 };
 Staircase.prototype.lock = function (stair) {
-  checkErr.ARG('lock', arguments, 1);
-  checkErr.UNDEFINED(this.stairs, stair);
+  this.checkErr.ARG('lock', arguments, 1);
+  this.checkErr.UNDEFINED(this.stairs, stair);
   this.stairs[stair].lock = true;
 };
 Staircase.prototype.unlock = function (stair) {
-  checkErr.ARG('unlock', arguments, 1);
-  checkErr.UNDEFINED(this.stairs, stair);
+  this.checkErr.ARG('unlock', arguments, 1);
+  this.checkErr.UNDEFINED(this.stairs, stair);
   this.stairs[stair].lock = false;
 };
 Staircase.prototype.isLocked = function (stair) {
-  checkErr.ARG('isLocked', arguments, 1);
-  checkErr.UNDEFINED(this.stairs, stair);
+  this.checkErr.ARG('isLocked', arguments, 1);
+  this.checkErr.UNDEFINED(this.stairs, stair);
   return this.stairs[stair].lock;
 };
 Staircase.prototype.setVal = function (stair, val) {
-  checkErr.ARG('setVal', arguments, 2);
-  checkErr.UNDEFINED(this.stairs, stair);
+  this.checkErr.ARG('setVal', arguments, 2);
+  this.checkErr.UNDEFINED(this.stairs, stair);
   this.stairs[stair].val[this.stairs[stair].val.length] = val;
 };
+Staircase.prototype.getReversals = function (stair) {
+    this.checkErr.ARG('getFinalVal', arguments, 1);
+    this.checkErr.UNDEFINED(this.stairs, stair);
+    // Find the inflection points
+    var reversals = []; // start with the first value
+    var direction = this.stairs[stair].direction*-1; // start the search getting harder
+    for(var i=1;i<this.stairs[stair].val.length;i++) {
+        if((direction==1 && this.stairs[stair].val[i]>this.stairs[stair].val[i-1]) ||
+            (direction==-1 && this.stairs[stair].val[i]<this.stairs[stair].val[i-1])) {
+            reversals[reversals.length] = this.stairs[stair].val[i-1]; // record reversal
+            direction = direction*-1; // reverse search direction
+        }
+    }
+    reversals.shift(); // first reversal doesn't count
+    return reversals;
+}
+Staircase.prototype.reversalLimitReached = function (stair) {
+    this.checkErr.ARG('getFinalVal', arguments, 1);
+    this.checkErr.UNDEFINED(this.stairs, stair);
+    var reversals = this.getReversals(stair);
+    return (reversals.length>=this.stairs[stair].reversalLimit && this.stairs[stair].reversalLimit!==0);
+}
+Staircase.prototype.getFinalVal = function (stair) {
+    this.checkErr.ARG('getFinalVal', arguments, 1);
+    this.checkErr.UNDEFINED(this.stairs, stair);
+    var reversals = this.getReversals(stair);
+    if(!reversals.length)
+        throw new Error("Staircase '"+this.stairs[stair].name+"': Not enough reversals to calculate final value.");
+    var sum = 0;
+    for(var i=0;i<reversals.length;i++)
+        sum = sum + reversals[i];
+    return sum/reversals.length; // Convergence value is the mean of the reversal points
+}
```

Comment:
I see that you are using these functions for your demo. Anyway it is true that it is a convenient way to retrieve these values